zerver/lib/rate_limiter.py

```diff
@@ -111,7 +112,7 @@ def rules(self) -> List[Tuple[int, int]]:
 
 class RateLimitedUser(RateLimitedObject):
     def __init__(self, user: UserProfile, domain: str = "api_by_user") -> None:
-        self.user = user
+        self.user = weakref.ref(user)
```

Comment:
Would it be simpler to sidestep the `SimpleLazyObject` issue by storing `user.id` and `user.rate_limits` in this class rather than `user`?

Reply:
Yeah, that sounds reasonable.